File: minesweeper/java-minesweeper/src/main/java/it/chicio/minesweeper/field/parser/FieldValidRowParser.java
package it.chicio.minesweeper.field.parser;

public class FieldValidRowParser implements FieldRowParser {
    private FieldRowContentParser fieldRowContentParser;

    public FieldValidRowParser(FieldRowContentParser fieldRowContentParser) {
        this.fieldRowContentParser = fieldRowContentParser;
    }

    public FieldsParsingStatus parse(String row, FieldsParsingStatus fieldsParsingStatus) throws RuntimeException {
        FieldsParsingStatus newFieldsParsingStatus = new FieldsParsingStatus(fieldsParsingStatus);
        if (isAValid(row)) {
            newFieldsParsingStatus.currentRowContent = row;
            return fieldRowContentParser.tryToParseRowAndUpdate(newFieldsParsingStatus);
        } else {
            return newFieldsParsingStatus;
        }
    }

    private boolean isAValid(String row) {
        return !row.equals("");
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/main/kotlin/it/chicio/minesweeper/field/parser/FieldValidRowParser.kt
package it.chicio.minesweeper.field.parser

class FieldValidRowParser(private val fieldRowContentParser: FieldRowContentParser): FieldRowParser {
    @Throws(RuntimeException::class)
    override fun parse(row: String?, fieldsParsingStatus: FieldsParsingStatus): FieldsParsingStatus =
            if (isAValid(row)) {
                val newFieldsParsingStatus = fieldsParsingStatus.copy()
                newFieldsParsingStatus.currentRowContent = row
                fieldRowContentParser.tryToParseRowAndUpdate(newFieldsParsingStatus)
            } else {
                fieldsParsingStatus.copy()
            }

    private fun isAValid(row: String?): Boolean = row != ""
}